pkg/registry/core/adapters/ns_registry.go

```diff
@@ -77,12 +98,21 @@ type networkServiceRegistryClient struct {
 }
 
 func (n *networkServiceRegistryClient) Register(ctx context.Context, in *registry.NetworkService, _ ...grpc.CallOption) (*registry.NetworkService, error) {
-	return n.server.Register(ctx, in)
+	doneCtx := withDone(ctx)
+	ns, err := n.server.Register(doneCtx, in)
+	if err != nil {
+		return nil, err
+	}
+	if !isDone(doneCtx) {
+		return ns, nil
+	}
+	return next.NetworkServiceRegistryClient(ctx).Register(ctx, in)
```

Comment:
We need to retrieve final context from server and use it instead of ctx, please see description in #366

Reply:
ok, fixed